wallet/txauthor/author.go

```diff
@@ -309,6 +323,54 @@ func spendWitnessKeyHash(txIn *wire.TxIn, pkScript []byte,
 	return nil
 }
 
+// spendTaprootKeyHash generates, and sets a valid witness for spending the
+// passed pkScript with the specified input amount. The input amount *must*
+// correspond to the output value of the previous pkScript, or else verification
+// will fail since the new sighash digest algorithm defined in BIP0341 includes
+// the input value in the sighash.
+func spendTaprootKeyHash(txIn *wire.TxIn, pkScript []byte,
+	inputValue int64, chainParams *chaincfg.Params, secrets SecretsSource,
+	tx *wire.MsgTx, hashCache *txscript.TxSigHashes, idx int) error {
+
+	// First obtain the key pair associated with this p2wkh address.
+	_, addrs, _, err := txscript.ExtractPkScriptAddrs(pkScript, chainParams)
+	if err != nil {
+		return err
+	}
+	privKey, _, err := secrets.GetKey(addrs[0])
+	if err != nil {
+		return err
+	}
+	pubKey := privKey.PubKey()
+
+	tapKey := txscript.ComputeTaprootKeyNoScript(pubKey)
+	p2trAddr, err := btcutil.NewAddressTaproot(
+		schnorr.SerializePubKey(tapKey), chainParams,
+	)
+	if err != nil {
+		return err
+	}
+
+	// With the concrete address type, we can now generate the
+	// corresponding witness program to be used to generate a valid witness
+	// which will allow us to spend this output.
+	witnessProgram, err := txscript.PayToAddrScript(p2trAddr)
+	if err != nil {
+		return err
+	}
+	witnessScript, err := txscript.TaprootWitnessSignature(
+		tx, hashCache, idx, inputValue, witnessProgram,
+		txscript.SigHashDefault, privKey,
```

Comment:
Was thinking about this when implementing, but is there any reason for us to _also_ generate 64 byte signatures when the sighash is all?

Reply:
Hmm, I'd say yes. Since this is an explicit call to `TaprootWitnessSignature`. Or what do you mean exactly?

Reply:
Basically to optimize the user's sig size for them behind the scenes. I updated the `RawTxInTaprootSignature` to _always_ append the sig type if not default in the latest version of the PR.